### PR TITLE
[276] Fix access requests logic

### PR DESCRIPTION
**Pull request by darokel**
### Context

https://trello.com/c/kUTUiG7r/276-fix-access-requests

### Changes proposed in this pull request

Fix the broken access request logic so recipient is actually linked up to requested providers.

### Guidance to review

- Create access request for a test user
- Approve and assert user is given access to the requested providers

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by darokel** [comment]
[Fix access requests](https://trello.com/c/kUTUiG7r/276-fix-access-requests)

**Review by defong — COMMENTED** [review]
Can we have test please.

**Comment by darokel** [comment]
> Can we have test please.

There's already a feature spec for this. The service also has an existing well covered unit test

**Comment by avinhurry** [comment]
> > Can we have test please.
> 
> There's already a feature spec for this. The service also has an existing well covered unit test

How come that didn't fail when it wasn't working? 🤔

**Comment by darokel** [comment]
@avinhurry @defong i've added an extra smoke check in the feature specs to make sure it does what it's supposed to do

File: spec/features/support/access_requests/managing_requests_spec.rb [hunk 1]
```diff
@@ -82,4 +84,12 @@ def and_i_should_see_steps_to_inform_the_publisher
   def admin
     @admin ||= create(:user, :admin)
   end
+
+  def access_request
+    @access_request ||= AccessRequest.first
+  end
+
+  def target_user
+    target_user ||= User.find_by(email: access_request.email_address.downcase)
```
Comment (defong) [inline spec/features/support/access_requests/managing_requests_spec.rb:93]:
```suggestion
    @target_user ||= User.find_by(email: access_request.email_address.downcase)
```

**Pull request merged**